flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -270,17 +270,13 @@ protected void rollback(FlinkResourceContext<CR> ctx) throws Exception {
 
         UpgradeMode upgradeMode = resource.getSpec().getJob().getUpgradeMode();
 
-        cancelJob(
-                ctx,
-                upgradeMode == UpgradeMode.STATELESS
-                        ? UpgradeMode.STATELESS
-                        : UpgradeMode.LAST_STATE);
+        cancelJob(ctx, upgradeMode);
```

Comment:
Why did you change this? We don't want to use a savepoint cancel during rollback

Reply:
I wanted to ensure that HA data is well deleted before to restoreJob for rollback also.

It looks to me that we should apply same pattern for upgrade and rollback relying only on the savepoint and ensuring HA data is here.
For ex. if the upgrade was linked to bump of flink version with some changes in HA data not compatible with rollbacked flink version

Reply:
Indeed it would be better
I will look into this and contact you through the original FLINK ticket